prov/rxm/src/rxm_cq.c

```diff
@@ -973,12 +985,17 @@ static inline ssize_t rxm_handle_atomic_resp(struct rxm_ep *rxm_ep,
 		ret = rxm_cq_tx_comp_write(rxm_ep,
 					   ofi_tx_cq_flags(tx_buf->pkt.hdr.op),
 					   tx_buf->app_context, tx_buf->flags);
-done:
-	if (tx_buf->pkt.hdr.atomic.op == ofi_op_atomic)
+
+	if (tx_buf->pkt.hdr.op == ofi_op_atomic) {
 		ofi_ep_wr_cntr_inc(&rxm_ep->util_ep);
-	else
+	} else if (tx_buf->pkt.hdr.op == ofi_op_atomic_compare ||
+		   tx_buf->pkt.hdr.op == ofi_op_atomic_fetch) {
 		ofi_ep_rd_cntr_inc(&rxm_ep->util_ep);
-
+	} else {
+		FI_WARN(&rxm_prov, FI_LOG_CQ, "unknown atomic request op!\n");
+		assert(0);
```

Comment:
Same thing here -- better way to bail on error?

Reply:
Ditto - see comment above.

Reply:
Like the above, this is an RXM internal protocol error if this scenario occurs. This is in the data path, so I think it should be considered if this needs to be explicitly checked in the non-debug path.